commit 829fb88ae677dd2ce99dbe1bbfbd1c43c1e7013b
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Mon Feb 20 08:59:16 2017 +0100

    mac80211: fix packet statistics for fast-RX
    
    commit 0328edc77d4f35014b35f32b46be0a7e16aae74f upstream.
    
    When adding per-CPU statistics, which added statistics back
    to mac80211 for the fast-RX path, I evidently forgot to add
    the "stats->packets++" line. The reason for that is likely
    that I didn't see it since it's done in defragmentation for
    the regular RX path.
    
    Add the missing line to properly count received packets in
    the fast-RX case.
    
    Fixes: c9c5962b56c1 ("mac80211: enable collecting station statistics per-CPU")
    Reported-by: Oren Givon <oren.givon@intel.com>
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index c073165..d3738b4 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -3953,6 +3953,7 @@ static bool ieee80211_invoke_fast_rx(struct ieee80211_rx_data *rx,
 	stats->last_rate = sta_stats_encode_rate(status);
 
 	stats->fragments++;
+	stats->packets++;
 
 	if (!(status->flag & RX_FLAG_NO_SIGNAL_VAL)) {
 		stats->last_signal = status->signal;
